src/mt.rs

```diff
@@ -67,20 +75,84 @@ impl Location {
     }
 }
 
+/// Configure a meta-tracer.
+pub struct MTBuilder {
+    hot_threshold: HotThreshold,
+}
+
+impl MTBuilder {
+    /// Create a meta-tracer with default parameters.
+    pub fn new() -> Self {
+        MTBuilder {
+            hot_threshold: DEFAULT_HOT_THRESHOLD,
+        }
+    }
+
+    /// Consume the `MTBuilder` and create a meta-tracer, returning the
+    /// [`MTThread`](struct.MTThread.html) representing the current thread.
+    pub fn init(self) -> MTThread {
+        MT::init(self.hot_threshold)
+    }
+
+    /// Change this meta-tracer builder's `hot_threshold` value.
+    pub fn hot_threshold(mut self, hot_threshold: HotThreshold) -> Self {
+        self.hot_threshold = hot_threshold;
+        self
+    }
+}
+
 /// A meta-tracer.
 pub struct MT {
     hot_threshold: HotThreshold,
+    active_threads: usize,
```

Comment:
This is a shame, but I appreciate it may be the only way to do it.

Did you try storing a list of references to join handles, so that the number of threads could be enumerated by asking the length of the list? I imagine the sharing and borrow checker could get hairy though?

Reply:
We don't have a `JoinHandle` for the main thread, so that doesn't quite work (at least not as-is).

Reply:
Good point.